mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/DockerJamesRule.java

```diff
@@ -42,7 +46,15 @@
     public DockerJamesRule(String image) {
         container = new DockerGenericContainer(image)
             .withExposedPorts(SMTP_PORT, IMAP_PORT)
-            .waitingFor(new HostPortWaitStrategy());
+            .waitingFor(new HostPortWaitStrategy())
+            .withLogConsumer(frame -> {
+                switch (frame.getType()) {
+                    case STDOUT:
+                        LOGGER.info(frame.getUtf8String());
+                    case STDERR:
+                        LOGGER.error(frame.getUtf8String());
```

Comment:
no default?

Reply:
What would you do if somebody add a new type of log?

Reply:
log to stdout?

Reply:
Doing nothing looks reasonable because you don't even know if there's a payload to read.